__tests__/endpoints.test.js

```diff
@@ -108,3 +110,62 @@ describe("GET /api/articles", () => {
       });
   });
 });
+
+describe("GET /api/articles/:article_id/comments", () => {
+  test("200: Should respond with an array of all comments for given article ID", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then(({ body }) => {
+        const comments = body.comments;
+        expect(Array.isArray(comments)).toBe(true);
+        expect(comments.length).toBeGreaterThan(0);
+        comments.forEach((comment) => {
+          expect(comment).toMatchObject({
+            body: expect.any(String),
+            votes: expect.any(Number),
+            author: expect.any(String),
+            article_id: expect.any(Number),
+            created_at: expect.any(String),
+          });
+          expect(comment).toHaveProperty("body");
+        });
+      });
+  });
+  test("200: comments should be returned sorted by most recent comments first", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then(({ body }) => {
+        const articleComments = body.comments;
+        expect(articleComments).toBeSorted({
+          key: "created_at",
+          descending: true,
+        });
+      });
+  });
+  test("400: to be 400 error code and message if endpoint is of wrong type or formatted incorrectly.", () => {
+    return request(app)
+      .get("/api/articles/not/comments")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Invalid article_id");
+      });
+  });
+  test("404: should return 404 error code and message if  endpoint is the correct type but doesnt have any results.", () => {
+    return request(app)
+      .get("/api/articles/5000/comments")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Not found");
+      });
+  });
+  test("404: should return a error code of 404 and a message of `ITS OVER 9000!!!` when given the correct type but doesnt have any results", () => {
+    return request(app)
+      .get("/api/articles/9001/comments")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("ITS OVER 9000!!!");
+      });
+  });
```

Comment:
I love it! ❤️ 

There is one test missing though and it is important. What if the article does exist but has no comments on it? Like article 13? This should return the empty rows array